### PR TITLE
Add subnet rescue command

### PR DESCRIPTION
**Pull request by pietrodimarco-dfinity**
This implements the command:
`dre subnet --id XXX  rescue --keep-nodes <nodes_features>`

### CODE REVIEW

File: rs/cli/src/cli.rs [hunk 1]
```diff
@@ -305,6 +305,13 @@ pub mod subnet {
             #[clap(long)]
             help_other_args: bool,
         },
+
+        /// Replace all nodes in a subnet
```
Comment (sasa-tomic) [inline rs/cli/src/cli.rs:309]:
```suggestion
        /// Replace all nodes in a subnet except those provided under "--keep-nodes"
```

File: rs/decentralization/src/network.rs [hunk 2]
```diff
@@ -758,18 +768,34 @@ pub trait TopologyManager: SubnetQuerier + AvailableNodesQuerier {
...
-        .with_exclude_nodes(exclude_nodes.clone())
-        .with_only_nodes_that_have_features(only_nodes.clone())
+        .including_from_available(NodeSelector::PrincipalIdList(include_nodes.clone()))
+        .excluding_from_available(NodeSelector::FeatureList(exclude_nodes.clone()))
+        .including_from_available(NodeSelector::FeatureList(only_nodes.clone()))
         .resize(size, 0)
     }
 }
 
+pub enum NodeSelector {
```
Comment (sasa-tomic) [inline rs/decentralization/src/network.rs:778]: I'm struggling to understand how `NodeSelector` works, and that's not a good sign.
Could you add some comments to make it easier for me? Please? :D

Comment (pietrodimarco-dfinity) [reply rs/decentralization/src/network.rs:806]: I added some examples and comments ;)

**Review by sasa-tomic — APPROVED** [review]
Awesome, thank you!

**Pull request merged**